tools/gendocs/main.go

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2023, Unikraft GmbH, The Cobra Authors and KraftKit Authors.
+// Licensed under the Apache-2.0 License (the "License").
+// You may not use this file except in compliance with the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	kraft "kraftkit.sh/internal/cli/kraft"
+
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	if len(os.Args[1:]) == 0 {
+		fmt.Printf("usage: %s outdir\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	outdir := os.Args[1]
+
+	if err := os.MkdirAll(outdir, 0o775); err != nil {
+		fmt.Println("could not create parent directories: %w", err)
+		os.Exit(1)
+	}
+
+	if err := generateMarkdown(kraft.NewCmd(), outdir); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func generateMarkdown(cmd *cobra.Command, dir string) error {
+	for _, c := range cmd.Commands() {
+		if !c.HasSubCommands() && (!c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand()) {
+			continue
+		}
+
+		if err := generateMarkdown(c, dir); err != nil {
+			return err
+		}
+	}
+
+	cmd.InitDefaultHelpCmd()
+	cmd.InitDefaultHelpFlag()
+
+	buf := new(bytes.Buffer)
+	name := cmd.CommandPath()
+
+	buf.WriteString("---\n")
+	buf.WriteString("title: \"" + name + "\"\n")
+	buf.WriteString("description: " + cmd.Short + "\n")
+	buf.WriteString("---\n\n")
+
+	if len(cmd.Long) > 0 {
+		if cmd == cmd.Root() {
+			buf.WriteString("```\n")
+		}
+		buf.WriteString(cmd.Long + "\n\n")
+		if cmd == cmd.Root() {
+			buf.WriteString("```\n")
+		}
+	}
+
+	if cmd.Runnable() {
+		buf.WriteString(fmt.Sprintf("```\n%s\n```\n\n", cmd.UseLine()))
+	}
+
+	if len(cmd.Example) > 0 {
+		buf.WriteString("### Examples\n\n")
+		buf.WriteString(fmt.Sprintf("```\n%s\n```\n\n", cmd.Example))
+	}
+
+	if err := printOptions(buf, cmd); err != nil {
+		return err
+	}
+
+	if hasSeeAlso(cmd) {
+		buf.WriteString("## See Also\n\n")
+		if cmd.HasParent() {
+			parent := cmd.Parent()
+			pname := parent.CommandPath()
+			link := pname + ".mdx"
```

Comment:
if it is a parent, you can just prefix with ../

```suggestion
			link := "../" pname + ".mdx"
```